src/core/line/line.ts

```diff
@@ -0,0 +1,44 @@
+import {
+    Directive,
+    Renderer,
+    ElementRef,
+    QueryList
+} from '@angular/core';
+
+/* Shared directive to count lines inside text */
+@Directive({ selector: '[md-line]' })
+export class MdLine {}
+
+/* Service that takes a query list of lines and sets the correct class on the host */
```

Comment:
I think technically this wouldn't be a "service" since it's not injected/injectable. Though that leads to the question- should this be injected instead of directly instantiated?